### ISSUE TITLE
Not working in Edge?

### ISSUE THREAD

**Issue by andrefgneves** [researchgate/react-intersection-observer#30]
Hi. I'm getting unexpected `intersectionRatio` values in Edge in an app I'm building and came here to check if someone else had the same issue. I then tried your storybook examples in Edge and looks like only the `Thresholds` and `Margin` examples seem to be working.

Here’s a video with the versions used: https://d.pr/v/kgwVnn

In my app, the issue is `intersectionRatio` coming back as `1` for a `1px` height element that is not within the viewport.

**Comment by Rendez** [comment]
To keep you posted:
- I was able to reproduce the issue only in Edge 16
- so far I found out that it's caused by the iframe of the storybook example
- opening the example outside an iframe works fine for Edge 16 (https://researchgate.github.io/react-intersection-observer/iframe.html?). Can you confirm that?

I tried simplying the CSS by removing flexbox and the rule: calc(100vh + 121px), but that's not the issue. I suspect the root of the IntersectionObserver object has to be either specified explicitely or this is a deeper bug in the spec. implementation. Could be a regression on IE.

I'll provide the final details when I have another moment.

**Comment by Rendez** [comment]
The BrowserStack version I tested matches quite closely your Edge Version:

![screen shot 2018-04-13 at 14 32 22](https://user-images.githubusercontent.com/61776/38735111-1fea8c68-3f28-11e8-82c3-a72835dc9c6f.png)

And like I said earlier, when I open the iframe link, it works normally for me:

![first](https://user-images.githubusercontent.com/61776/38735253-94cf30d8-3f28-11e8-9be0-c08f73a46461.gif)

However, in the demo page, seems like the examples that do not contain the prop `rootMargin` nor 'thresholds` suffer from this issue while the others do not:

![second](https://user-images.githubusercontent.com/61776/38735618-8bc2636a-3f29-11e8-9ed7-8192f76405d8.gif)

I'm won't be at work for a couple of weeks, I would be happy if you could provide your last feedback and perhaps you feel encouraged to find out on your end and submit a PR?

Here are the Edge issues related to the IO: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/?page=1&q=intersection

Either it's the native implementation of Edge 16 or something else is fishy.

Ultimately I'll forward this bug to my team.

Cheers

**Comment by Rendez** [comment]
@andrefgneves I got a machine at the office running Edge 16 and locally it's also buggy for me within the iframe, unless I set some negative rootMargin, e.g.: `-12px 0px`.

Someone opened an issue for the Edge's team: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/17319665/

To me they seem related since the `intersectionRatio` are correct but the `isIntersecting` calculation is off. Previous Edge versions didn't have this bug.

There's nothing I can do to circumvent this issue, so I recommend you include the polyfill for Edge16 until the issue is resolved.
